# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/x86-shared/CodeGenerator-x86-shared.cpp
# Commit: 38d4e11b069b
# Full Hash: 38d4e11b069b609199ed6017ce80e199d2c5773b
# Author: Lars T Hansen <lhansen@mozilla.com>
# Date: 2020-05-20 15:28:23
# Regressor Bug: 1631228
# File Overlap Count: 1
# Description:
#   Bug 1631228 - wasm ion simd, part 3: ion code generation. r=bbouvier
#   
#   Add SIMD opcodes to WasmIonCompile and implement code generation for
#   them on x64.
#   
# ==============================================================================

diff -r 9aacd7b8b252 -r 38d4e11b069b js/src/jit/x86-shared/CodeGenerator-x86-shared.cpp
--- a/js/src/jit/x86-shared/CodeGenerator-x86-shared.cpp	Wed May 20 07:02:49 2020 +0000
+++ b/js/src/jit/x86-shared/CodeGenerator-x86-shared.cpp	Wed May 20 07:03:54 2020 +0000
@@ -281,6 +281,11 @@
       case MIRType::Float32:
         masm.storeFloat32(ToFloatRegister(ins->arg()), dst);
         return;
+#ifdef ENABLE_WASM_SIMD
+      case MIRType::Simd128:
+        masm.storeUnalignedSimd128(ToFloatRegister(ins->arg()), dst);
+        return;
+#endif
       default:
         break;
     }